presto-hive/src/main/java/com/facebook/presto/hive/parquet/ParquetPageSourceFactory.java

```diff
@@ -240,4 +245,19 @@ private static boolean columnTypeSupported(List<HiveColumnHandle> columns)
 
         return regularColumnsSupported && partitionColumnsSupported;
     }
+
+    private static boolean isTypeSupported(TypeSignature typeSignature)
+    {
+        if (MAP.equals(typeSignature.getBase()) || ARRAY.equals(typeSignature.getBase())) {
+            return false;
+        }
+        else if (ROW.equals(typeSignature.getBase())) {
+            return typeSignature.getParameters().stream()
+                .map(TypeSignatureParameter::getTypeSignatureOrNamedTypeSignature)
+                .filter(Optional::isPresent)
+                .map(Optional::get)
+                .allMatch(ParquetPageSourceFactory::isTypeSupported);
+        }
+        return true;
```

Comment:
It looks like the code supports all primitive types now, but say it didn't support `DATE`, then this code would allow that though.